### PR TITLE
numba implementation of dream stepper

### PR DESCRIPTION
**Pull request by pkienzle**
Minimally tested.

Need to check performance relative to compiled C and pure python versions.

### CODE REVIEW

**Comment by bmaranville** [comment]
Seems like the C DLL will still be useful in extremely parallel environments - I don't see a good reason to get rid of it, unless we move the step logic into the parallelizer as well, as you suggested.  It's just a source file and a small set of compile instructions.

**Comment by pkienzle** [comment]
It's hard to write a test to show that the compiled c and the numba are doing the same thing. Too many calls to the random number generator.

File: bumps/dream/diffev.py [hunk 1]
```diff
@@ -109,22 +167,27 @@ def de_step(Nchain, pop, CR, max_pairs=2, eps=0.05,
         else:
             raise RuntimeError("Select failed...should never happen")
 
-        # If no step was specified (exceedingly unlikely!), then
```
Comment (bmaranville) [inline bumps/dream/diffev.py:112]: Does this affect the stability if merged?  Will we see cases where no step was specified, and what will happen if we do?

Comment (bmaranville) [reply bumps/dream/diffev.py:112]: After speaking with @pkienzle it seems like this won't be an issue

**Pull request merged**